core/common/src/main/java/alluxio/Configuration.java

```diff
@@ -406,10 +408,35 @@ public static Duration getDuration(PropertyKey key) {
   }
 
   /**
-   * @return a view of the internal {@link Properties} of as an immutable map
+   * @return a map of the resolved properties(include default properties)
    */
   public static Map<String, String> toMap() {
-    return Collections.unmodifiableMap(PROPERTIES);
+    // Resolve values with ${VALUE} format
+    Map<String, String> map = toRawMap();
+    String nestedPropIdentifier = "$";
```

Comment:
can we re-use the nested property parsing already built into this class? `Configuration.get(String)` will do the recursive lookups needed to resolve nested properties. I'm not sure `StrSubstitutor` will match our lookup logic exactly.

Reply:
Will change to 
``` java
if (value.contains(nestedPropIdentifier)) {
        map.put(entry.getKey(), lookup(value));
}
```
lookup will Lookup key names to handle ${key} stuff and return resolved string value.